changelog/fragments/upgrade-kb.yaml

```
@@ -0,0 +1,57 @@
+# entries is a list of entries to include in
+# release notes and/or the migration guide
+entries:
+  - description: >
+      Upgrade controller-runtime (sigs.k8s.io/controller-runtime) dependency version
+      from `v0.6.2` to `v0.6.3`. More info: https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.6.3
+
+    # kind is one of:
+    # - addition
+    # - change
+    # - deprecation
+    # - removal
+    # - bugfix
+    kind: "change"
+
+    # Is this a breaking change?
+    breaking: false
+
+    # Migration can be defined to automatically add a section to
+    # the migration guide. This is required for breaking changes.
+    migration:
+      header: Upgrade controller-runtime version
+      body: >
+        In the `go.mod` file replace `sigs.k8s.io/controller-runtime v0.6.2`
+        with `sigs.k8s.io/controller-runtime v0.6.3` and then run `go mod tidy`.
+  - description: >
+      Upgrade the Kubebuilder CLI dependecy version from `f7a3b65dd250` to `c993a2a221fe`.
```

Operator projects don't directly depend on `kubebuilder`, so this is rather confusing. We probably don't need to notify users about `operator-sdk` _binary_ dependency changes (those that are not scaffolded) unless they affect commands that have a runtime (ex. k8s versions).

How about
```suggestion
      Fix scaffolded (multi-group only) `controllers/<group>/suite_test.go` to have
      a correct `envtest.Environment.CRDDirectoryPaths` ([kubebuilder#1665](https://github.com/kubernetes-sigs/kubebuilder/pull/1665)).
```

---

Then, we came back for the first version of this PR where all changes/bugfixes that impact the end-user was described on it. see:  https://github.com/operator-framework/operator-sdk/pull/4062/commits/59cb16829753e40525be3773b0a44f75851e732b

Note that it was a suggestion made by @joelanford

---

Do you think users care about what version of kubebuilder the `operator-sdk` binary uses, or whether a scaffold fix occurred? You can still link the commit range without discussing it.

---

> Do you think users care about what version of kubebuilder the operator-sdk binary uses, or whether a scaffold fix occurred? 

I do not think so. Because of this my first version of this PR just add fragment for all bugfixes changes that are visible to the SDK end-users

> You can still link the commit range without discussing it.
What is your suggestion? How do you think that we can do that?

---

I will apply your specific suggestion let's see.